src/azul/deployment.py

```diff
@@ -85,20 +86,45 @@ def api_gateway_export(self, gateway_id):
         return json.load(response['body'])
 
     @property
-    def es_endpoint(self) -> Netloc:
+    def es_endpoint(self) -> Optional[Netloc]:
         if config.es_endpoint:
             return config.es_endpoint
         else:
-            es_domain_status = self.es.describe_elasticsearch_domain(DomainName=config.es_domain)
-            return es_domain_status['DomainStatus']['Endpoint'], 443
+            return self._describe_es_domain(config.es_endpoint)
 
     @property
-    def es_instance_count(self) -> int:
+    def es_instance_count(self) -> Optional[int]:
         if config.es_endpoint:
             return config.es_instance_count
         else:
+            return self._describe_es_domain(config.es_instance_count)
+
+    def _describe_es_domain(self, es_config_param) -> Union[Netloc, int, None]:
```

Comment:
Too complicated. I don't want to give too much away because I want you to think harder about this. I think you'll learn more from that than if I held your hand through the refactoring.